cpp/kiss_icp/core/Registration.cpp

```diff
@@ -83,10 +83,10 @@ Sophus::SE3d AlignClouds(const std::vector<Eigen::Vector3d> &source,
             auto Weight = [&](double residual2) { return square(th) / square(th + residual2); };
             auto &[JTJ_private, JTr_private] = J;
             for (auto i = r.begin(); i < r.end(); ++i) {
-                const auto &[J_r, r] = compute_jacobian_and_residual(i);
-                const double w = Weight(r.squaredNorm());
+                const auto &[J_r, residual] = compute_jacobian_and_residual(i);
```

Comment:
Shadowing the `r` parameter of the lambda here is something that MSVC complained about.

Reply:
Damn 🤔 why is this even working? I'll need to look further into this